src/macros.jl

```diff
@@ -286,22 +286,26 @@ end
 """
     macro_assign_and_return(code, variable, name;
                             final_variable=variable,
-                            register_fun::Union{Nothing, Function}=nothing,
+                            register::Bool=false,
                             model=nothing)
 
 Return runs `code` in a local scope which returns the value of `variable`
 and then assign `final_variable` to `name`.
-If `register_fun` is given, `register_fun(model, name, variable)` is called.
+If `register` is true is given, `register_object(model, name, variable)` is
+called in the generated code.
 """
 function macro_assign_and_return(code, variable, name;
                                  final_variable=variable,
-                                 register_fun::Union{Nothing, Function}=nothing,
+                                 register::Bool=false,
```

Comment:
Maybe we can remove the `register` boolean argument and branch on whether `model` is `nothing` or not ?
This function does not work if `register` is `true` and `model` is nothing anyway so it might be cleaner